src/components/ContactForm/ContactForm.tsx

```diff
@@ -1,30 +1,139 @@
 import * as React from "react";
-
+import Recaptcha from "react-google-invisible-recaptcha";
+import { Button } from "../Button";
 import * as styles from "./ContactForm.module.scss";
 
 interface Props {
     style?: React.CSSProperties;
     className?: string;
 }
 interface State {
-    myStateValue?: boolean;
+    name: string;
+    email: string;
+    phone: string;
+    message: string;
 }
-
+const RECAPTCHA_KEY = process.env.GATSBY_SITE_RECAPTCHA_KEY;
 const INITIAL_STATE: State = {
-    myStateValue: true
+    name: "",
+    email: "",
+    phone: "",
+    message: ""
+};
+
+const encode = (data: { [key: string]: string }) => {
+    return Object.keys(data)
+        .map(
+            key => encodeURIComponent(key) + "=" + encodeURIComponent(data[key])
+        )
+        .join("&");
 };
 
 export class ContactForm extends React.PureComponent<Props, State> {
     public state = INITIAL_STATE;
+    public recaptcha?: Recaptcha;
+    public handleChange = (
+        e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) =>
+        this.setState({ [e.target.name]: e.target.value } as Pick<
+            State,
+            keyof State
+        >);
+
+    public handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        if (!this.recaptcha) {
+            console.error("Missing recaptcha");
+            return;
+        }
+        this.recaptcha.execute();
+    };
+
+    public handleCaptcha = () => {
+        console.log("handle captcha");
```

Comment:
There are console logs that probably shouldn't be in here...

Reply:
This is still a WIP, which I was planning to drop anyway. Certainly not ready for review.

Reply:
okay... I was't certain as it had been PRd

Reply:
I use "WIP" in the title to mean "please ignore". It's only a PR so that it would build to Netlify so I could test the form handling.